### DISCUSSION TITLE
Is it possible to update ui.expansion() text?

### DISCUSSION THREAD

**Discussion by frankhuurman** [zauberzeug/nicegui#2199] [Ideas / Feature Requests]
### Question

For our application at work we want to have a column of a lot of different expansion elements with each their own fields like this:
![image](https://github.com/zauberzeug/nicegui/assets/19980966/c7fca24a-d660-4e65-a858-38da224661fa)

Everything works, expansion units can be added to/deleted from the column list and each unit has their own respective input fields.

But I'm wondering if the actual text label of an **ui.expansion** element can be updated after creation?
I've tried everything from value binding to set_text() or .text() or ._text with an .update() call after it but nothing seems to be working so I'm unsure if this even has an implementation.

Preferably the text would be updated by the changing text value of the first ui.input() inside of the expansion element but for now I'm looking for any way to make it work :)

Thanks in advance!

**Comment by rodja** [comment]
Try this:

```py
def update_title():
    ex._props['label'] = 'new title'
    ex.update()

ex = ui.expansion('test')
ui.button('update title', on_click=update_title)
```

**Reply by falkoschindler** [reply]
Or using `.props()`:
```py
ex = ui.expansion('test')
ui.button('update title', on_click=lambda: ex.props('label="new title"'))
```

**Comment by falkoschindler** [comment]
Even though there are ways to update the text (see https://github.com/zauberzeug/nicegui/discussions/2199#discussioncomment-7881371), I guess we could make `ui.expansion` a `TextElement` with bindable text. Therefore I'll convert this discussion into a feature request.
